lua/entities/gmod_wire_expression2/core/custom/lifesupport.lua

```diff
@@ -282,8 +293,14 @@ e2function number entity:lsUnlinkAll()
 
 	local canTool = this:CPPICanTool(self.player, "toolgun") and 1 or 0
 	if canTool then
-		local netid = ls_get_ent_netid(this)
-		RD.UnlinkAllFromNode(netid)
+		if this.IsPump then
```

Comment:
I think you should leave `lsUnlinkAll` alone. Your change makes it only unlink a pump if a pump is targeted, which is not what it is supposed to do :p

Reply:
Isn't a pump only able to connect to a node? So in theory we would want this logic to run?

Reply:
@KieranFYI `lsUnlinkAll`, is supposed to, from any entity, grab its LS network and unlink everything on that network. So you target 1 storage, it grabs its node, and unlinks everything from that node.

Reply:
If you look at the reload option for the link tool, it does the same logic, just in a different order.

https://github.com/SpaceAgeMP/spacebuild/blob/main/lua/weapons/gmod_tool/stools/rd3_dev_link.lua#L269

Reply:
@KieranFYI The R on the link tool also only does the "unlink all" codepath when targeting a node. When targeting a normal LS entity, it only unlinks that entity.

Also looking at that tool, damn that's a lot of special cases...